File: src/test/java/com/roscopeco/jasm/model/DoubleMathTests.java
package com.roscopeco.jasm.model;

public interface DoubleMathTests {
    float testD2f(final double in);
    int testD2i(final double in);
    long testD2l(final double in);

    double testDadd(final double d1, final double d2);
    double testDconst0();
    double testDconst1();
    int testDcmpg(final double a, final double b);
    int testDcmpl(final double a, final double b);

    double testDdiv(final double a, final double b);
    double testDmul(final double a, final double b);
    double testDneg(final double a);
    double testDrem(final double a, final double b);
    double testDsub(final double a, final double b);
}

File: src/test/kotlin/com/roscopeco/jasm/model/DoubleMathTests.kt
package com.roscopeco.jasm.model

interface DoubleMathTests {
    fun testD2f(input: Double): Float
    fun testD2i(input: Double): Int
    fun testD2l(input: Double): Long
    fun testDadd(d1: Double, d2: Double): Double
    fun testDconst0(): Double
    fun testDconst1(): Double
    fun testDcmpg(a: Double, b: Double): Int
    fun testDcmpl(a: Double, b: Double): Int
    fun testDdiv(a: Double, b: Double): Double
    fun testDmul(a: Double, b: Double): Double
    fun testDneg(a: Double): Double
    fun testDrem(a: Double, b: Double): Double
    fun testDsub(a: Double, b: Double): Double
}